src/core/balance-sheet.ts

```diff
@@ -1,36 +1,50 @@
 import { getBalance as getBitcoindBalance } from "@services/bitcoind"
 import { lndsBalances } from "@services/lnd/utils"
 import { baseLogger } from "@services/logger"
-import { InvoiceUser, User } from "@services/mongoose/schema"
 import { ledger } from "@services/mongodb"
 
 import { WalletFactory } from "./wallet-factory"
 import { runInParallel } from "./utils"
+import { WalletInvoicesRepository } from "@services/mongoose"
+import { RepositoryError } from "@domain/errors"
+import { User } from "@services/mongoose/schema"
 
 const logger = baseLogger.child({ module: "balanceSheet" })
 
-export const updatePendingLightningTransactions = async () => {
-  // select distinct user ids from pending invoices
-  const usersWithPendingInvoices = InvoiceUser.aggregate([
-    { $match: { paid: false } },
-    { $group: { _id: "$uid" } },
-  ])
-    .cursor({ batchSize: 100 })
-    .exec()
+const isRepoError = (obj): boolean => obj instanceof RepositoryError
```

Comment:
If you want to use this in multiple places perhaps it should go in `src/domain/utils.ts`

Reply:
done